python/graphstorm/dataloading/dataset.py

```diff
@@ -553,6 +554,13 @@ def __init__(self, graph_name, part_config, train_etypes, eval_etypes=None,
                                                  decoder_edge_feat,
                                                  lm_feat_ntypes=lm_feat_ntypes,
                                                  lm_feat_etypes=lm_feat_etypes)
+        if self._train_etypes == [DEFAULT_ETYPE]:
+            assert self._g.ntypes == [DEFAULT_NTYPE] and \
+                   self._g.etypes == [DEFAULT_ETYPE[1]], \
+                f"It is required to be a homogeneous graph when not providing " \
+                f"target_etype on node task, expect node type {[DEFAULT_NTYPE]} and " \
```

Comment:
Why node task needs target_etype?

Reply:
For homogeneous graph, it is strictly required for both of the node type and the edge type, so it is required to check both node type and edge type meet the requirement.

Reply:
Meanwhile, code here is for GSgnnEdgeTrainData.